src/client.js

```diff
@@ -147,66 +129,44 @@ function parseImageInfo(info) {
-                .then(reply => {
-                    const immages = JSON.parse(reply);
-                    const images = {};
-                    const promises = [];
-
-                    for (const image of immages || []) {
-                        images[image.Id] = image;
-                        promises.push(podmanCall("libpod/images/" + image.Id + "/json", "GET", {}, system));
-                    }
-
-                    Promise.all(promises)
-                            .then(replies => {
-                                for (const reply of replies) {
-                                    const info = JSON.parse(reply);
-                                    images[info.Id] = Object.assign(images[info.Id], parseImageInfo(info));
-                                    images[info.Id].isSystem = system;
-                                }
-                                resolve(images);
-                            })
-                            .catch(reject);
-                })
-                .catch(reject);
-    });
+    const options = {};
+    if (id)
+        options.filters = JSON.stringify({ id: [id] });
+    return podmanJson("libpod/images/json", "GET", options, system)
+            .then(reply => {
+                const images = {};
+                const promises = [];
+
+                for (const image of reply || []) {
+                    images[image.Id] = image;
+                    promises.push(podmanJson("libpod/images/" + image.Id + "/json", "GET", {}, system));
+                }
+
+                return Promise.all(promises)
+                        .then(replies => {
+                            for (const info of replies) {
+                                images[info.Id] = Object.assign(images[info.Id], parseImageInfo(info));
+                                images[info.Id].isSystem = system;
+                            }
+                            return images;
+                        });
+            });
 }
 
 export function getPods(system, id) {
-    return new Promise((resolve, reject) => {
-        const options = {};
-        if (id)
-            options.filters = JSON.stringify({ id: [id] });
-        podmanCall("libpod/pods/json", "GET", options, system)
-                .then(reply => resolve(JSON.parse(reply)))
-                .catch(reject);
-    });
+    const options = {};
+    if (id)
+        options.filters = JSON.stringify({ id: [id] });
+    return podmanJson("libpod/pods/json", "GET", options, system);
 }
 
-export function delImage(system, id, force) {
-    return new Promise((resolve, reject) => {
-        const options = {
-            force,
-        };
-        podmanCall("libpod/images/" + id, "DELETE", options, system)
-                .then(reply => resolve(JSON.parse(reply)))
-                .catch(reject);
-    });
-}
+export const delImage = (system, id, force) => podmanJson("libpod/images/" + id, "DELETE", { force }, system);
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1339-20230706-033702-672c013e-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)